apis/lambda/v1alpha1/custom_types.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+
+// CustomCodeSigningConfigParameters includes custom fields for CodeSigningConfigParameters.
+type CustomCodeSigningConfigParameters struct{}
+
+// CustomFunctionParameters includes custom fields for FunctionParameters.
+type CustomFunctionParameters struct {
+
+	// SecurityGroupIDs is the list of IDs for the SecurityGroups.
+	SecurityGroupIDs []string `json:"vpcConfig.securityGroupIDs,omitempty"`
+
+	// SecurityGroupIDRefs is a list of references to SecurityGroups used to set
+	// the SecurityGroupIDs.
+	// +optional
+	SecurityGroupIDRefs []xpv1.Reference `json:"vpcConfig.securityGroupIdRefs,omitempty"`
+
+	// SecurityGroupIDsSelector selects references to SecurityGroupID used
+	// to set the SecurityGroupIDs.
+	// +optional
+	SecurityGroupIDSelector *xpv1.Selector `json:"vpcConfig.securityGroupIdSelector,omitempty"`
+
+	// SubnetIDs is the list of IDs for the Subnets.
+	SubnetIDs []string `json:"subnetIds,omitempty"`
+
+	// SubnetIDRefs is a list of references to Subnets used to set
+	// the SubnetIDs.
+	// +optional
+	SubnetIDRefs []xpv1.Reference `json:"vpcConfig.subnetIdRefs,omitempty"`
+
+	// SubnetIDsSelector selects references to Subnets used
+	// to set the SubnetIDs.
+	// +optional
+	SubnetIDSelector *xpv1.Selector `json:"vpcConfig.subnetIdSelector,omitempty"`
+
+	// KMSKeyARN is the arn of the KMS Key.
+	// +immutable
+	KMSKeyARN *string `json:"kmsKeyARN,omitempty"`
```

Comment:
```suggestion
	// KMSKeyARN is the arn of the KMS Key.
	// +immutable
	KMSKeyARN *string `json:"kmsKeyARN,omitempty"`
```

We can use the generated one in the reference resolver.